lib/rubygems/specification.rb

```diff
@@ -2454,7 +2457,9 @@ def to_ruby
       next if handled.include? attr_name
       current_value = self.send(attr_name)
       if current_value != default_value(attr_name) || self.class.required_attribute?(attr_name)
-        result << "  s.#{attr_name} = #{ruby_code current_value}" unless current_value.is_a?(OpenSSL::PKey::RSA)
+        unless defined?(OpenSSL) && current_value.is_a?(OpenSSL::PKey::RSA)
```

Comment:
```suggestion
        unless has_openssl && current_value.is_a?(OpenSSL::PKey::RSA)
```

Reply:
I like this approach

Reply:
i prefer keeping consistent with the library by checking for the constant